### ISSUE TITLE
WebSocket improvement: Notifications API

### ISSUE THREAD

**Issue by starlocke** [sj26/mailcatcher#342]
It would be great if a new message, appearing automatically via WebSockets, also triggers notifications via the Notifications API.

https://developer.mozilla.org/en-US/docs/Web/API/Notifications_API

Cautionary: There should also be some kind of grouping/throttling for massive batches that arrive all at once.

**Comment by sj26** [comment]
Yep, it'd be great to add notifications! The javascript is a bit of a mess, though. I've been working on React-ifying it with GraphQL including subscriptions which would groove nicely with notifications. I'll take this a feature request for then.
